agent/utils.mjs

```diff
@@ -0,0 +1,659 @@
+        } else {
+            throw new Error(
+                "UTF-8 decode: code point 0x" +
+                c.toString( 16 ) + " exceeds UTF-16 reach" );
+        }
+    }
+    return s;
+}
+
+export function hexToBytes( strHex, isInversiveOrder ) { // convert a hex string to a byte array
+    isInversiveOrder = !!(
+        ( isInversiveOrder != null && isInversiveOrder != undefined && isInversiveOrder )
+    );
+    strHex = strHex || "";
+    strHex = "" + strHex;
+    strHex = strHex.trim().toLowerCase();
+    if( strHex.length > 1 && strHex[0] == "0" && ( strHex[1] == "x" || strHex[1] == "X" ) )
+        strHex = strHex.substr( 2, strHex.length - 2 );
+    if( ( strHex.length & 1 ) !== 0 )
+        strHex = "0" + strHex;
+    const cnt = strHex.length;
+    let i, j;
+    const arrBytes = new Uint8Array( cnt / 2 );
+    for( i = 0, j = 0; i < cnt; ++j, i += 2 )
+        arrBytes[j] = parseInt( strHex.substr( i, 2 ), 16 );
+    if( isInversiveOrder )
+        invertArrayItemsLR( arrBytes );
+    return arrBytes;
+}
+
+export function bytesToHex( arrBytes, isInversiveOrder ) { // convert a byte array to a hex string
+    isInversiveOrder = !!(
+        ( isInversiveOrder != null && isInversiveOrder != undefined && isInversiveOrder )
+    );
+    const hex = [];
+    for( let i = 0; i < arrBytes.length; i++ ) {
+        const current = arrBytes[i] < 0 ? arrBytes[i] + 256 : arrBytes[i];
+        const c0 = ( current >>> 4 ).toString( 16 );
+        const c1 = ( current & 0xF ).toString( 16 );
+        if( isInversiveOrder ) {
+            hex.splice( 0, 0, c0 );
+            hex.splice( 1, 0, c1 );
+        } else {
+            hex.push( c0 );
+            hex.push( c1 );
+        }
+    }
+    return hex.join( "" );
+}
+
+export function bytesAlignLeftWithZeroes( arrBytes, cntMin ) {
+    const arrOneZeroByte = new Uint8Array( 1 );
+    arrOneZeroByte[0] = 0;
+    while( arrBytes.length < cntMin )
+        arrBytes = bytesConcat( arrOneZeroByte, arrBytes );
+    return arrBytes;
+}
+
+export function bytesAlignRightWithZeroes( arrBytes, cntMin ) {
+    const arrOneZeroByte = new Uint8Array( 1 );
+    arrOneZeroByte[0] = 0;
+    while( arrBytes.length < cntMin )
+        arrBytes = bytesConcat( arrBytes, arrOneZeroByte );
+    return arrBytes;
+}
+
+export function concatTypedArrays( a, b ) { // a, b TypedArray of same type
+    if( typeof a == "string" )
+        a = hexToBytes( a );
+    if( typeof b == "string" )
+        b = hexToBytes( b );
+    const c = new ( a.constructor )( a.length + b.length );
+    c.set( a, 0 );
+    c.set( b, a.length );
+    return c;
+}
+
+export function concatByte( ui8a, byte ) {
+    const b = new Uint8Array( 1 );
+    b[0] = byte;
+    return concatTypedArrays( ui8a, b );
+}
+
+export function bytesConcat( a1, a2 ) {
+    a1 = a1 || new Uint8Array();
+    a2 = a2 || new Uint8Array();
+    return concatTypedArrays( a1, a2 );
+}
+
+export function toArrayBuffer( buf ) {
```

Comment:
Check here to use standard functions instead of loops
https://stackoverflow.com/questions/37228285/uint8array-to-arraybuffer

Reply:
`toArrayBuffer` is not needed at all. So I will remove it.